Common/FileHandlers/FileReadHandler.cs

```diff
@@ -24,11 +25,11 @@ public async Task<string> ReadAllTextFromFile(string name, string extension, str
         }
     }
 
-    public bool FileExists(string name, string extension, string location)
+    public bool FileExists(string name, FileExtension extension, string location)
     {
-        if (name == string.Empty || extension == string.Empty) { throw new ArgumentNullException("Name, Text or Extension provided is empty"); }
+        if (name == string.Empty) { throw new ArgumentNullException("Name, Text or Extension provided is empty"); }
 
-        name += extension;
+        name += "." + extension.ToString().ToLower();
```

Comment:
Make a createFileName method

Reply:
Make the method in a base FileHandler class so you can use it in both FileReadHandler and FileWriteHandler

Reply:
Make a UnitTest for it